### PR TITLE
Use `:textual?` flag for Search in Files

### LINKED ISSUE #7586: IDE: Exclude binary files from the search.
**Is your feature request related to a problem? Please describe (REQUIRED):**
We have more and more files in the project (3400 files, 180 megabytes) and the Ctrl/Cmd+F project search function is getting slower and slower. Why? The IDE seems to search through all files, even binary ones.

For example, I entered "IHDR" (PNG header) and the IDE found this text in GLBs:

image

**Describe the solution you'd like (REQUIRED):**
Improve the detection of binary files or exclude ".glb" from the search.

### PR DESCRIPTION
**Pull request by vlaaad**
Technical note: we now look at `:textual?` flag of a resource type to determine if a resource without save data might be textual.

User-facing changes:

We no longer search through `.glb` files when performing Search in Files.

Fixes #7586

### CODE REVIEW

File: editor/src/clj/editor/defold_project_search.clj [hunk 1]
```diff
@@ -51,7 +51,9 @@
   [{:keys [content resource] :as save-data}]
   (or (when (some? content)
         (make-line-coll #(BufferedReader. (StringReader. content))))
-      (when (and (resource/exists? resource) (not (text-util/binary? resource)))
+      (when (and (resource/exists? resource)
+                 (resource/textual? resource)
+                 (not (text-util/binary? resource)))
```
Comment (matgis) [inline editor/src/clj/editor/defold_project_search.clj:56]: If `(:textual? resource-type)` is `true`, we don't need to check if `(text-util/binary? resource)`, right? If I understand it correctly, we only need to do that if `resource-type` is `nil`?

**Review by matgis — CHANGES_REQUESTED** [review]
I don't think we need to call `text-util/binary?` for known resource types?

File: editor/src/clj/editor/defold_project_search.clj (hunk 1 above)
Comment (vlaaad) [reply editor/src/clj/editor/defold_project_search.clj:56]: Hmm, if there is no resource type, `resource/textual?` will return true, but I see what you mean, we shouldn't check the contents of resources with specified types...

**Review by matgis — APPROVED** [review]
Looks good! 👍

**Pull request merged**